tiledb/sm/query/result_space_tile.h

```diff
@@ -48,30 +50,25 @@ namespace sm {
  * @tparam The datatype of the array domain.
  */
 template <class T>
-struct ResultSpaceTile {
-  /** The (global) coordinates of the first cell in the space tile. */
-  std::vector<T> start_coords_;
-
-  /**
-   * A vector of pairs `(fragment id, fragment domain)`, sorted on
-   * fragment id in descending order. Note that only fragments
-   * with domains that intersect this space tile will be included
-   * in this vector.
-   */
-  std::vector<std::pair<unsigned, const T*>> frag_domains_;
-
-  /**
-   * The (dense) result tiles for this space tile, as a map
-   * `(fragment id) -> (result tile)`.
-   */
-  std::map<unsigned, ResultTile> result_tiles_;
-
+class ResultSpaceTile {
+ public:
   /** Default constructor. */
   ResultSpaceTile() = default;
 
   /** Default destructor. */
   ~ResultSpaceTile() = default;
 
+  ResultSpaceTile(
```

Comment:
I don't see this constructor used anywhere in this patch, which is fine. If are using this (or intend to use it in the near future), we should consider a constructor with universal references for the STL arguments (e.g. `std::vector<T>&& start_coords`) so that we can `move` these STL containers into this class without the alloc/copy/free overhead on the STL container's internal heaped memory.

Reply:
It is used in the unit tests, but good catch. I will remove and use the setters instead.